kratos/utilities/particles_utilities.h

```diff
@@ -0,0 +1,330 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <iostream>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "utilities/binbased_fast_point_locator.h"
+#include "utilities/atomic_utilities.h"
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Collection of utilities to compute statistic of particles
+/** Detail class definition.
+*/
+class ParticlesUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ParticlesUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(ParticlesUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    ParticlesUtilities(){};
+
+    /// Destructor.
+    virtual ~ParticlesUtilities(){};
+
+    template<unsigned int TDim>
+    static void CountParticlesInNodes(
+            BinBasedFastPointLocator<TDim>& rLocator,
+            ModelPart& rVolumeModelPart,
+            const ModelPart& rParticlesModelPart,
+            const Variable<double>& rCounterVariable
+            )
+        {
+
+            //reset the EMBEDDED_VELOCITY
+            block_for_each(rVolumeModelPart.Nodes(), [&rCounterVariable](auto& rNode){
+                rNode.GetValue(rCounterVariable) = 0.0;
```

Comment:
In here you rely that a `SetValue(rCounterVariable)` has been already performed. Maybe you can do the `SetValue` in here just in case. Not sure, just pointing it out.

Reply:
ok